starlite/openapi/controller.py

```diff
@@ -58,13 +59,40 @@ def get_schema_from_request(request: Request) -> "OpenAPI":
             An [OpenAPI][pydantic_openapi_schema.v3_1_0.open_api.OpenAPI] instance.
 
         Raises:
-            [ImproperlyConfiguredException][starlite.exceptions.ImproperlyConfiguredException]: If the application
-                `openapi_schema` attribute is `None`.
+            ImproperlyConfiguredException: If the application `openapi_config` attribute is `None`.
         """
         if not request.app.openapi_schema:  # pragma: no cover
             raise ImproperlyConfiguredException("Starlite has not been instantiated with OpenAPIConfig")
         return request.app.openapi_schema
 
+    def should_serve_endpoint(self, request: Request) -> bool:
+        """This method verifies that the requested path is within the enabled
+        endpoints in the openapi_config.
+
+        Args:
+            request: To be tested if endpoint enabled.
+
+        Returns:
+            A boolean.
+
+        Raises:
+            ImproperlyConfiguredException: If the application `openapi_config` attribute is `None`.
+        """
+        if not request.app.openapi_config:  # pragma: no cover
+            raise ImproperlyConfiguredException("Starlite has not been instantiated with OpenAPIConfig")
+
+        request_path = set(filter(None, request.url.path.split("/")))
+        root_path = set(filter(None, self.path.split("/")))
+
+        config = request.app.openapi_config
+        if request_path == root_path and config.root_schema_site in config.enabled_endpoints:
+            return True
+
+        if request_path & config.enabled_endpoints:
```

Comment:
I suggest to `or` these two conditions instead of having two if clauses. Alternatively, save them into variables with clear names and then check `if <var1> or <var2>:`

Reply:
I'm not a big fan of this idea, I prefer separated and simple if statements. It's easier to read and understand imho

Reply:
as you see fit